core/modules/filter/filter.admin.inc

```diff
@@ -14,17 +14,53 @@
  */
 function filter_admin_overview($form) {
   // Overview of all formats.
-  $formats = filter_formats();
+  $formats = filter_formats(NULL, TRUE, TRUE);
   $fallback_format = filter_fallback_format();
   $editors = filter_get_editors();
 
+  $form['help'] = array(
+    '#type' => 'help',
+    '#markup' => t('The order of the text formats here reflects the order in which they will show up in the <em>Editor</em> selector when creating content (subject to the text formats to which a user has access.'),
+    '#weight' => -100,
+  );
+
   $form['#tree'] = TRUE;
+
   foreach ($formats as $id => $format) {
+    // Prepare roles for each fomrat.
+    $roles = array();
+    foreach (filter_get_roles_by_format($format) as $role_name) {
+      $role = user_role_load($role_name);
+      $roles[] = check_plain($role->label);
+    }
+    $roles_markup = $roles ? theme('item_list', array('items' => $roles)) : t('No roles specified for this format.');
+
+    // Prepare operations links for each format.
     $links = array();
+
     $links['configure'] = array(
       'title' => t('Configure'),
       'href' => "admin/config/content/formats/$id",
     );
+
+    if ($format->status) {
+      $links['disable'] = array(
+        'title' => t('Disable'),
+        'href' => "admin/config/content/formats/$id/disable",
```

Comment:
what you have here is very slightly slower to process than `'admin/config/content/formats/' . $id . '/disable'`. Let's be as consistent as we can in core, and choose the faster way :)